roofit/roofitcore/test/TestStatistics/testLikelihoodGradientJob.cpp

```diff
@@ -467,10 +458,7 @@ TEST_P(LikelihoodGradientJob, Gaussian1DAlsoWithLikelihoodJob)
    std::tie(nll, pdf, data, values) = generate_1D_gaussian_pdf_nll(w, 10000);
    RooRealVar *mu = w.var("mu");
 
-   RooArgSet *savedValues = dynamic_cast<RooArgSet *>(values->snapshot());
-   if (savedValues == nullptr) {
-      throw std::runtime_error("params->snapshot() cannot be casted to RooArgSet!");
-   }
+   RooArgSet savedValues = *values->snapshot();
```

Comment:
```suggestion
   RooArgSet savedValues;
   values->snapshot(savedValues);
```
I'm sorry, I see now that the previous change request I made was not completely clear!

Not only "getParameters", but also "snapshot" returns an owning pointer. It would be better to use the overload of "snapshot" that takes an output parameter, so you can do everything on the stack without memory leak.

Same goes for all the other calls of "snapshot()" in this file.

Reply:
Done.